### ISSUE TITLE
Add "x-ob-url" header dynamically in IG

### ISSUE THREAD

**Issue by mariantiris** [SecureApiGateway/SecureApiGateway#216]
The x-ob-url header must be provided to the resource server. This URL will be added by the resource server on the "Self" claims on the response. Currently this header is hardcoded in postman.

**To Reproduce**
Send a valid resource access request without the header from postman (for example /accounts). The error thrown by the resource server is: 

`{"Code":"OBRI.Request.Invalid","Id":"4ff9f7f5-eda2-44ba-b98d-57ee15fce891","Message":"An error happened when parsing the request arguments","Errors":[{"ErrorCode":"UK.OBIE.Header.Missing","Message":"Missing request header 'x-ob-url' for method parameter of type String","Path":null,"Url":null}]}`

**Expected behaviour**
IG should build the header formed with the following rule:

`request.getHeaders().get("X-Forwarded-Host") + request.path`

**Current behaviour**
The requests fail if the header is not manually added on the request from postman or any other tool.

**Comment by mariantiris** [comment]
@BohoCode, @jorgesanchezperez  need your opinion on this. Would it be better to build the links by using the rs.fqdn, or an incoming request header?

**Comment by BohoCode** [comment]
@mariantiris Alternatively could the X-Forwarded-Host be used in the RS instead of the x-ob-uri?

**Comment by dbadham-fr** [comment]
I think it would be better to avoid relying on any headers not documented in the OB spec where possible. 

Even though `X-Forwarded-Host` is a defacto standard, it is not documented in the spec, also the `x-ob-url ` header appears to be something we have invented. Any deviations from the spec would need to be documented and maintained by us, which I think we would rather avoid.

Therefore I think the configuration based approach is better.

**Comment by dbadham-fr** [comment]
This is causing the FAPI test suite to fail, as the test suite implements the OB spec and therefore does not pass the `x-ob-url`

**Comment by BohoCode** [comment]
So the x-ob-uri is required by the RS and is used to build the self links required in the response. 

In old projects it was provided by the gateway using;
`new ServletServerHttpRequest(request).getURI().toString()`

This would
> Return the URI of the request (including a query string if any, but only if it is well-formed for a URI representation).
Which I take to mean it would return the full URI including path.

**Comment by dbadham-fr** [comment]
I've raised a separate issue to unblock the FAPI suite, this will set the header within IG.
https://github.com/SecureBankingAccessToolkit/SecureBankingAccessToolkit/issues/672

This issue can remain open in order to figure out and implement the best solution to the issue.

**Comment by dbadham-fr** [comment]
Fixed in #1035 

Controllers in the RS now use Spring HATEOAS to create self links, the `x-ob-url` header can now be removed from the IG routes.
